tensorflow/python/training/optimizer.py

```diff
@@ -282,8 +282,14 @@ def apply_gradients(self, grads_and_vars, global_step=None, name=None):
     # This is a default implementation of apply_gradients() that can be shared
     # by most optimizers.  It relies on the subclass implementing the following
     # methods: _create_slots(), _prepare(), _apply_dense(), and _apply_sparse().
+
+    grads_to_tensors_list = []
+    vars_list = []
     grads_and_vars = tuple(grads_and_vars)  # Make sure repeat iteration works
```

Comment:
I think this line is redundant now. (It was previously necessary in order to be able to iterate over the argument twice, when the argument was a generator expression. However, you now only iterate over it once.)